libcontainer/rootfs_linux.go

```diff
@@ -572,6 +572,11 @@ func pivotRoot(rootfs, pivotBaseDir string) error {
 	if err != nil {
 		return fmt.Errorf("can't create pivot_root dir %s, error %v", pivotDir, err)
 	}
+	defer func() {
+		if errVal := os.Remove(pivotDir); errVal != nil && err == nil {
+			err = errVal
```

Comment:
@mikebrow i would not cancatinate because we will lose information.  its better to return the original error so that things like os.IsExit() or finding the specific syscall.Errno is correct.  The Remove() error is not as important as the original error returned via this function